lnrpc/walletrpc/walletkit.proto

```diff
@@ -350,6 +350,169 @@ service WalletKit {
+    required for any on-chain related transactions or messages.
+    */
+    rpc SignCoordinatorStreams (stream SignCoordinatorResponse)
+        returns (stream SignCoordinatorRequest);
+}
+
+message SignCoordinatorRequest {
+    /*
+    A unique request ID of a SignCoordinator gRPC request. Useful for mapping
+    requests to responses.
+    */
+    uint64 request_id = 1;
+
+    /*
+    Messages between the watch-only node and the remote signer can only be of
+    certain types.
+    */
+    oneof sign_request_type {
+        /*
+        The registration complete message signals that the watch-only lnd has
+        registered the remote signers handshake, and accepts the remote signer
+        connection.
+        */
+        bool registration_complete = 2;
+
+        /*
+        To ensure that the remote signer is still active and alive, the
+        watch-only lnd can send a Ping message to the remote signer, which
+        should then respond with the respective Pong message.
+        */
+        bool ping = 3;
+
+        /*
+        Requests a shared public key from the remote signer.
+        */
+        signrpc.SharedKeyRequest shared_key_request = 4;
+
+        /*
+        Requests that the remote signer signs the passed message.
+        */
+        signrpc.SignMessageReq sign_message_req = 5;
+
+        /*
+        Requests a MuSig2 Session of the remote signer.
+        */
+        signrpc.MuSig2SessionRequest mu_sig2_session_request = 6;
+
+        /*
+        Requests that the remote signer registers a nonce with the referenced
+        MuSig2 Session.
+        */
+        signrpc.MuSig2RegisterNoncesRequest mu_sig2_register_nonces_request = 7;
+
+        /*
+        Requests that the remote signer signs the passed message digest with
+        the referenced MuSig2 Session.
+        */
+        signrpc.MuSig2SignRequest mu_sig2_sign_request = 8;
+
+        /*
+        Requests that the remote signer combines and adds the passed partial
+        signatures for the referenced MuSig2 Session.
+        */
+        signrpc.MuSig2CombineSigRequest mu_sig2_combine_sig_request = 9;
+
+        /*
+        Requests that the remote signer removes/cleans up the referenced
+        MuSig2 session.
+        */
+        signrpc.MuSig2CleanupRequest mu_sig2_cleanup_request = 10;
+
+        /*
+        Requests that the remote signer signs the passed PSBT.
+        */
+        SignPsbtRequest sign_psbt_request = 11;
+    }
+}
+
+message SignCoordinatorResponse {
+    /*
+    The request ID this response refers to.
+    */
+    uint64 ref_request_id = 1;
+
+    /*
+    The remote signer responses can only be of certain types.
+    */
+    oneof sign_response_type {
+        /*
+        The Signer Registration message is sent by the remote signer when it
+        connects to the watch-only lnd node, to initialize a handshake between
+        the nodes.
+        */
+        bool signer_registration = 2;
```

Comment:
Let me know if it makes sense to add some kind of versioning for the signer, so that the watch-only node can know what kind of functionality the signer supports.

Reply:
Hmm, good question. I don't think that adding an independent version here gives us much. I think (at least in the beginning), we'd want the signer and watch-only to be on the same main `lnd` version. So maybe we should send the full version string here and initially just compare it to the one running on the watch-only, then error out if it differs?
Or don't do anything yet other than specifying this field as a string, so we can do it in the future?

Reply:
> Or don't do anything yet other than specifying this field as a string, so we can do it in the future?

Makes sense, thanks! I agree that it probably makes sense to just enable the support so that we can define the versioning idea when we actually decide on it in the future, so I edited this to be a string instead.